### DISCUSSION TITLE
Is there a option to do a window selection for capture?

### DISCUSSION THREAD

**Discussion by ghxstweb** [flameshot-org/flameshot#3478] [General]
Flameshot is a great alternative to KDE's spectacle, and I believe it actually functions more efficiently. However, I'm missing the feature that enables me to screenshot the entire window of an app, and I was wondering if you all happen to know if there is an option or a way I can utilize a "window" screenshotting feature. Thanks!

**Comment by mmahmoudian** [comment]
In Wayland there is not, but in X11 it is easy to do. For the X11 just checkout the "select window" function of this small project of mine:

https://codeberg.org/mehrad/dmenu_shot

basically these are the lines you should read:
https://codeberg.org/mehrad/dmenu_shot/src/commit/426d6390508f3e9d3759d82757d5a115dae7aa1f/dmenu_shot.sh#L207-L221


For wayland we have multiple problems:
1. not every wayland implementation is the same and provide the same API
2. if we want to avoid using wayland APIs for window detection, we have to do image processing to find the window. You can [search for "opencv"](https://github.com/flameshot-org/flameshot/issues?q=opencv+) and follow discussions there.
